app/src/main/java/com/example/fragmentassg2/fragments/FragmentLogin.kt

```diff
@@ -58,11 +68,14 @@ class FragmentLogin : Fragment(), View.OnClickListener {
     private fun login(){
         val number ="9953755955"
 
-        if(editPhone.editText?.text.toString()==number){
+        val checkPhoneNumber:Boolean=viewModel.validatePhoneNumber(number)
+
+        if(editPhone.editText?.text.toString()==number && checkPhoneNumber){
```

Comment:
@raghav32 the check for the text and number is equal or not should also go inside viewmodel, Ideally you should call a function inside viewmodel which will do all the business logic and returns true or false & on the basis of that you should show relevant view to user